unlock-app/src/components/interface/checkout/main/Select.tsx

```diff
@@ -225,16 +226,35 @@ export function Select({ checkoutService }: Props) {
               props.network || paywallConfig.network || 1
 
             const lockData = await web3Service.getLock(lock, networkId)
+
+            let price
+
+            if (account) {
+              price = await web3Service.purchasePriceFor({
```

Comment:
This _may_ fail in some cases (for example when the lock uses a hook like the password hook). In that case, please make show the "default" price which is the `lockData.keyPrice`